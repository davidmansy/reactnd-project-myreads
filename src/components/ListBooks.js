import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import Bookshelf from './Bookshelf';

function ListBooks(props) {
  const { shelves, books, handleShelfChange } = props;
  return (
    <div className="list-books">
      {/*TODO Replace by a header component*/}
      <div className="list-books-title">
        <h1>MyReads</h1>
      </div>
      <div className="list-books-content">
        {/*TODO Add a view to handle an empty shelf*/}
        <div>
          {shelves.map(shelf => (
            <Bookshelf
              key={shelf.id}
              shelf={shelf}
              books={books}
              handleShelfChange={handleShelfChange}
            />
          ))}
        </div>
      </div>
      <div className="open-search">
        <Link to="/search">Add a book</Link>
      </div>
    </div>
  );
}

export default ListBooks;
