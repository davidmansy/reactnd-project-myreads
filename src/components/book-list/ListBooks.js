import React from 'react';
import ListBooksHeader from './ListBooksHeader';
import Bookshelves from './Bookshelves';
import OpenSearch from '../book-search/OpenSearch';
import PropTypes from 'prop-types';

const BOOK_HEADER_TITLE = 'MyReads';

function ListBooks(props) {
  const { books, handleShelfChange, isLoading, searchPath } = props;

  return (
    <div className="list-books">
      <ListBooksHeader title={BOOK_HEADER_TITLE} />
      <Bookshelves
        books={books}
        handleShelfChange={handleShelfChange}
        isLoading={isLoading}
      />
      <OpenSearch searchPath={searchPath} />
    </div>
  );
}

ListBooks.propTypes = {
  books: PropTypes.arrayOf(PropTypes.object).isRequired,
  handleShelfChange: PropTypes.func.isRequired,
  isLoading: PropTypes.bool.isRequired,
  searchPath: PropTypes.string.isRequired
};

export default ListBooks;
