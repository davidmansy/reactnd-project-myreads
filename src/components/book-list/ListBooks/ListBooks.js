import React from 'react';
import ListBooksHeader from '../ListBooksHeader/ListBooksHeader';
import Bookshelves from '../Bookshelves/Bookshelves';
import OpenSearch from '../../book-search/OpenSearch/OpenSearch';
import PropTypes from 'prop-types';

const BOOK_HEADER_TITLE = 'MyReads';
const ADD_BOOK_CONTROL_LABEL = 'Add a book';

function ListBooks(props) {
  const { books, handleShelfChange, isLoading, searchPath } = props;

  return (
    <div className="list-books">
      <ListBooksHeader title={BOOK_HEADER_TITLE} />
      <Bookshelves
        books={books}
        handleShelfChange={handleShelfChange}
        isLoading={isLoading}
      />
      <OpenSearch
        searchPath={searchPath}
        addControlLabel={ADD_BOOK_CONTROL_LABEL}
      />
    </div>
  );
}

ListBooks.propTypes = {
  books: PropTypes.arrayOf(PropTypes.object).isRequired,
  handleShelfChange: PropTypes.func.isRequired,
  isLoading: PropTypes.bool.isRequired,
  searchPath: PropTypes.string.isRequired
};

export default ListBooks;
