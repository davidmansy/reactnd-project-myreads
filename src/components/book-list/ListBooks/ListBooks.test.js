import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import ListBooks from './ListBooks';
import { MemoryRouter } from 'react-router-dom';
import { cleanup } from 'react-testing-library';

afterEach(cleanup);

describe('ListBooks', () => {
  const props = {
    books: [],
    handleShelfChange: () => {},
    isLoading: false,
    searchPath: '/search'
  };

  it('renders', () => {
    const container = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <ListBooks {...props} />
      </MemoryRouter>,
      container
    );
  });

  it('snapshots', () => {
    const testRenderer = TestRenderer.create(
      <MemoryRouter>
        <ListBooks {...props} />
      </MemoryRouter>
    );
    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
