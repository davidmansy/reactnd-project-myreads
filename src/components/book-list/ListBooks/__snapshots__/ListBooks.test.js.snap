// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListBooks snapshots 1`] = `
<div
  className="list-books"
>
  <div
    className="list-books-title"
  >
    <h1>
      MyReads
    </h1>
  </div>
  <div
    className="list-books-content"
  >
    <div
      className="bookshelves"
    >
      <div
        className="bookshelf"
      >
        <h3
          className="bookshelf-title"
        >
          Want to Read
        </h3>
        <div
          className="bookshelf-books"
        >
          <div
            className="bookshelf-empty"
          >
            There are not books available in this shelf
          </div>
        </div>
      </div>
      <div
        className="bookshelf"
      >
        <h3
          className="bookshelf-title"
        >
          Currently Reading
        </h3>
        <div
          className="bookshelf-books"
        >
          <div
            className="bookshelf-empty"
          >
            There are not books available in this shelf
          </div>
        </div>
      </div>
      <div
        className="bookshelf"
      >
        <h3
          className="bookshelf-title"
        >
          Read
        </h3>
        <div
          className="bookshelf-books"
        >
          <div
            className="bookshelf-empty"
          >
            There are not books available in this shelf
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="open-search"
  >
    <a
      href="/search"
      onClick={[Function]}
    >
      Add a book
    </a>
  </div>
</div>
`;
